mitodl/odl-video-service
# Download videos to Dropbox

#### What are the relevant tickets?
Closes #261

#### What's this PR do?
- Adds a popup menu to video cards, as described in #202 (thanks @gsidebo for doing most of the work on this).
- Adds a 'Save To Dropbox' button to video detail pages and 'Save To Dropbox' menu option to video cards, which opens a Dropbox Saver widget allowing the user to pick a Dropbox folder to save the original source video file to.

#### How should this be manually tested?
- From a collection detail page:
  - There should be a menu icon on each video card.
  - Click the video card menu icon.  If you are an admin for the collection, there should be 3 choices: Share, Edit, Save To Dropbox.
  - Click the 'Save To Dropbox' menu option.  A dropbox widget should appear.  The filename to save should equal the title of the video plus original source extension (if not already in the title).
  - Choose a folder to save it to and click 'Save'
  - After a while verify that the file is in the chosen Dropbox folder.
- From a video detail page:
  - If you are an admin, there should be share, edit, and dropbox buttons directly under the video title.
  - Click the 'Save To Dropbox' button.  The dropbox widget should appear.  Repeat steps from above to transfer the video to dropbox.
